frame/vamm/src/lib.rs

```diff
@@ -649,55 +654,58 @@ pub mod pallet {
 			ensure!(!Self::is_vamm_closed(&vamm_state, &None), Error::<T>::VammIsClosed);
 
 			match asset_type {
-				AssetType::Base => Ok(DecimalOf::<T>::from_inner(vamm_state.base_asset_twap)),
-				AssetType::Quote => Ok(DecimalOf::<T>::from_inner(vamm_state.quote_asset_twap)),
+				AssetType::Base => Ok(vamm_state.base_asset_twap),
+				AssetType::Quote => Ok(vamm_state.quote_asset_twap),
 			}
 		}
 
-		/// Updates the time weighted average price of the desired asset.
+		/// Updates the time weighted average price of both assets.
 		///
 		/// # Overview
 		/// In order for the caller to update the time weighted average price of
-		/// the desired asset, it has to request it to the Vamm Pallet. The
-		/// pallet will perform the needed sanity checks and update the runtime
-		/// storage with the desired twap value, returning it in case of
-		/// success.
+		/// the assets, it has to request it to the Vamm Pallet. The pallet will
+		/// perform the needed sanity checks and update the runtime storage with
+		/// the desired twap values, returning both it in case of success.
```

Comment:
returning both it in case of success. -> returning both **in the** case of success.

Reply:
Ok, thanks for the correction! :)